### ISSUE TITLE
`http` example fails to compile as of 0.15.1

### ISSUE THREAD

**Issue by kevinpoitra** [antoyo/relm#155]
Trying to compile the example results in these two error messages:

```
error[E0277]: `*mut HttpMsg` cannot be sent between threads safely
   --> src\main.rs:245:36
    |
245 |             connect_async!(client, connect_to_host_async(host, 80), relm, Connection);
    |                                    ^^^^^^^^^^^^^^^^^^^^^ `*mut HttpMsg` cannot be sent between threads safely
    |
    = help: within `[closure@<::relm::macros::connect_async macros>:7:25: 8:73 stream:relm::EventStream<HttpMsg>]`, the trait `std::marker::Send` is not implemented for `*mut HttpMsg`
    = note: required because it appears within the type `std::marker::PhantomData<*mut HttpMsg>`
    = note: required because it appears within the type `relm::EventStream<HttpMsg>`
    = note: required because it appears within the type `[closure@<::relm::macros::connect_async macros>:7:25: 8:73 stream:relm::EventStream<HttpMsg>]`

error[E0277]: `*mut HttpMsg` cannot be sent between threads safely
   --> src\main.rs:259:44
    |
259 |                     connect_async!(writer, write_async(buffer.into_bytes(), PRIORITY_DEFAULT), self.model.relm,
    |                                            ^^^^^^^^^^^ `*mut HttpMsg` cannot be sent between threads safely
    |
    = help: within `[closure@<::relm::macros::connect_async macros>:7:25: 8:73 stream:relm::EventStream<HttpMsg>]`, the trait `std::marker::Send` is not implemented for `*mut HttpMsg`
    = note: required because it appears within the type `std::marker::PhantomData<*mut HttpMsg>`
    = note: required because it appears within the type `relm::EventStream<HttpMsg>`
    = note: required because it appears within the type `[closure@<::relm::macros::connect_async macros>:7:25: 8:73 stream:relm::EventStream<HttpMsg>]`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
error: Could not compile `http`.`
```

I can confirm that it compiles just fine with 0.15.0.

**Comment by antoyo** [comment]
Fixed by #157.
Thanks for reporting this issue.

**Comment by kevinpoitra** [comment]
Thank you for the fix! I did just notice that it seems like it's not totally fixed though. In a program I'm working on, I have some code that is based off of the http example itself. In the example, you do:

```Rust
connect_async!(client, connect_to_host_async(&host, 80), relm, Connection);
```
which compiles fine. In my program, I want to be notified if there was an error while creating the connection, so I use:
```Rust
connect_async!(client, connect_to_host_async(&host, 80), relm, Connection, ConnectionFailed);
```
which does not compile. It throws this error:
```
error[E0599]: no method named `into_inner` found for type `std::result::Result<gio::SocketConnection, gdk::Error>` in the current scope
  --> src\http\mod.rs:93:9
   |
93 |         connect_async!(client, connect_to_host_async(&host, 80), relm, Connection, ConnectionFailed);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

**Comment by antoyo** [comment]
I made a typo, sorry about that.
It should be fixed in 4ec198ce6123b67c6604d627b267caf8e5f8a99e.
